### DISCUSSION TITLE
Terraform Cloud Remote 환경에서의 각 환경별(Dev, Stage, Prod) tfvars 설정 관련해서 질분 드립니다.

### DISCUSSION THREAD

**Discussion by b100to** [tedilabs/fastcampus-devops#63] [Q&A] [answered]
안녕하세요. 강의 영상보고 실무에 많은 도움을 받고 있습니다.
다름이 아닌 TFC사용 하면서 지속적으로 골머리를 앓고 있는 것이 있습니다.
Local 환경에서는 tfvars를 통해 환경을 분배 시킬 수 있는데(ex. dev.tfvars, stage.tfvars, prod.tfvars) TFC에서는 tfvars를 지원해주지 않습니다.
*.auto.tfvars를 사용하게 되면 분배를 시킬 수가 없고 UI 환경에서 variables를 할당하기엔 비효율적인 부분이 많습니다.

병진님 또한 TFC remote 환경을 사용중이라고 말씀하셨는데요, 어떤 방식으로 tfvars를 혹은 이와 다른 방법을 어떻게 사용하고 계신지 궁금합니다.

현재 같은 경우엔 Dev 환경이면 'TFC에 TF_CLI_ARGS_plan = var-file=dev.tfvars'  와 같이 TFC 내에 환경변수를 지정하여 사용중입니다. 허나 Secret.tfvars도 사용하고 싶고 그런데 하나의 환경변수만 지정할 수 있어 자유롭지 못합니다.

https://www.reddit.com/r/Terraform/comments/qhosyq/terraform_cloud_workflow_with_environment/

위 방법도 괜찮긴 한데 괜찮은 방법인지 또 사용방법도 좀 어려워서 다른 분의 의견도 궁금하기에 질문드리게 되었습니다.

감사합니다.

**Comment by posquit0** [comment, marked as answer]
안녕하세요~!

강의가 도움이 되고 있다니 기쁩니다. ㅎㅎ
저는 TFC를 테라폼 상태 저장소 및 모듈 관리 목적으로만 사용 중에 있습니다! TFC의 실행 모드를 remote가 아닌 local 모드로 사용합니다.
아마 강의 중에서도 동일하게 언급을 하였던 것 같은데.. remote를 사용하고 있다고 언급이 되었다면 실수가 있었던 것 같습니다... 😭 

만약 제가 remote 모드를 사용하게 된다면 [TFE Provider](https://registry.terraform.io/providers/hashicorp/tfe/latest)를 이용하여 TFC의 워크스페이스와 변수 모두 코드로 같이 관리할 것 같습니다.
또, 제가 선호하는 방향은 워크스페이스의 변수를 tfvars로 관리하는 것이 아닌 YAML 파일로 관리하는 방법입니다. [tedilabs/github#config.yaml](https://github.com/tedilabs/github/blob/main/repositories/config.yaml)에서 확인할 수 있는 것처럼요. 동일 디렉토리 내에서 환경을 구분하게 된다면 YAML 파일을 환경 별로 만들고 워크스페이스마다 적절한 YAML 파일을 참조하도록 구성을 할 것 같습니다.

감사합니다! :)

**Reply by b100to** [reply]
정말 빠른 답변 감사합니다... 기존에 remote 모드를 사용중이여서 변경하는데 있어 고민이 많았는데, local 모드와 아틀란티스를 고려해보도록 하겠습니다. 감사합니다. :)

또한 밑에 언급하신 내용 처럼 config.yaml 방식도 감사합니다. 재 확인 후 질문이 있으면 또 질문 남겨 보도록 하겠습니다 감사합니다!
